sources/packages/web/src/views/aest/student/Applications.vue

```diff
@@ -17,5 +22,19 @@ export default {
       required: true,
     },
   },
+  setup(_props: any) {
+    const router = useRouter();
+
+    const goToApplication = (id: number) => {
+      return router.push({
+        name: AESTRoutesConst.APPLICATION_DETAILS,
+        params: {
+          applicationId: id,
+          studentId: _props.studentId,
+        },
+      });
+    };
+    return { goToApplication };
+  },
 };
```

Comment:
router.push is a promise. You may need to use async/await in this function.

Reply:
As per Vue, if we are waiting for the result of navigation which can be (NavigationFailure | void | undefined ) and do something with the result then await is useful. IMO otherwise it should be fine without it. @ann-aot @andrewsignori-aot thoughts?

Reply:
I believe, if are doing anything after the router, then we need to wait, else I think there is no need. correct me if I am wrong @dheepak-aot @henryoforeh-aot  @andrewsignori-aot 
![image](https://user-images.githubusercontent.com/77353155/155206585-07c4b936-9db2-4075-a5b8-87c148627c66.png)
https://router.vuejs.org/guide/advanced/navigation-failures.html